power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -116,23 +123,29 @@ template <typename T, dataset_type_tag dataset_type> class ColumnarAttributeRang
         std::span<AttributeBuffer<Data> const> attribute_buffers_{};
     };
 
-    class iterator : public boost::iterator_facade<iterator, T, boost::random_access_traversal_tag, Proxy, Idx> {
+    class iterator
+        : public IteratorFacade<iterator, std::conditional_t<is_data_mutable_v<dataset_type>, Proxy, Proxy const>,
+                                Idx> {
       public:
-        using value_type = Proxy;
+        using value_type = std::conditional_t<is_data_mutable_v<dataset_type>, Proxy, Proxy const>;
+        using difference_type = Idx;
```

Comment:
Why do we need this, instead of using `Idx` directly?

Reply:
`value_type` of a iterator should not be `const`. Even for a const iterator.

See
https://stackoverflow.com/questions/12819405/why-is-stditerator-traitsvalue-type-non-const-even-for-a-const-iterator

Reply:
good one. will add `reference_type` instead. i was struggling with this indeed.

Reply:
done